statsmodels/tsa/automatic/sarimax.py

```diff
@@ -0,0 +1,18 @@
+"""automatic forecasting for SARIMAX models."""
+import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+
+def auto_order(endog, criteria='aic', d=0, max_order=(3, 3), spec=None):
+    """Auto order function for SARIMAX models."""
+    aic_matrix = np.zeros(p, q)
+    for p in range(max_order[0]):
+        for q in range(max_order[1]):
+            if p == 0 and q == 0:
```

Comment:
You don't need to skip this case, since if `exog` is set in `spec`, then linear regression is the special case `p=0, q=0`.

Reply:
Yaa, I agree but if we skip this case, then the SARIMAX model fails as one of the states should be positive.
I think we should put a check for this

Reply:
If this is true, can you file a bug report with code to replicate? When I try, it seems to work:

```python
import pandas as pd
import statsmodels.api as sm

dta = sm.datasets.macrodata.load_pandas().data
dta.index = pd.PeriodIndex(start='1959Q1', end='2009Q3', freq='Q')

mod = sm.tsa.SARIMAX(dta['infl'], order=(0, 0, 0))
res = mod.fit()
print(res.summary())
```

```
                           Statespace Model Results                           
==============================================================================
Dep. Variable:                   infl   No. Observations:                  203
Model:                        SARIMAX   Log Likelihood                -619.610
Date:                Thu, 17 May 2018   AIC                           1241.220
Time:                        07:51:27   BIC                           1244.533
Sample:                    03-31-1959   HQIC                          1242.560
                         - 09-30-2009                                         
Covariance Type:                  opg                                         
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
sigma2        26.2234      2.466     10.632      0.000      21.389      31.058
===================================================================================
Ljung-Box (Q):                      712.11   Jarque-Bera (JB):                60.66
Prob(Q):                              0.00   Prob(JB):                         0.00
Heteroskedasticity (H):               0.45   Skew:                             0.74
Prob(H) (two-sided):                  0.00   Kurtosis:                         5.23
===================================================================================
```

Reply:
In that case,  I'll go check this code again. I'll take this into account and edit the code for necessary changes.

Reply:
It throws a ValueError for the number of states to be positive which I have pasted in 
https://gist.github.com/abhijeetpanda12/e2d2c14744c84b368910842a4ecfe0f7

Reply:
Can you do:

```python
import statsmodels.api as sm.
print(sm.version.version)
```

You seem to be using an old version, whereas you should be developing against master.

Reply:
I was using version 0.8. I have now rebased my local repository to the current master running version 0.10
Thanks for helping me out.